corehq/apps/analytics/tasks.py

```diff
@@ -754,3 +889,63 @@ def _max_edition():
         '{}is_on_extended_trial_plan'.format(env): _is_on_extended_trial(),
         '{}max_edition_of_paying_plan'.format(env): _max_edition()
     }
+
+
+@periodic_task(run_every=crontab(minute="0", hour="7"), queue='background_queue')
+def cleanup_blocked_hubspot_contacts():
+    """
+    Remove any data stored about users from blocked domains and email domains
+    from Hubspot in case it somehow got there.
+    :return:
+    """
+    if not HUBSPOT_ENABLED:
+        return
+
+    # First delete any user information from users that are members of
+    # blocked domains
+    blocked_domains = get_blocked_hubspot_domains()
+    for domain in blocked_domains:
+        user_query = UserES().domain(domain).source(['email', 'username'])
+
+        total_users = user_query.count()
+        chunk_size = 30  # Hubspot recommends fewer than 100 emails per request
+        num_chunks = int(math.ceil(float(total_users) / float(chunk_size)))
+
+        for chunk in range(num_chunks):
+            blocked_users = (user_query
+                             .size(chunk_size)
+                             .start(chunk * chunk_size)
+                             .run()
+                             .hits)
+            blocked_emails = []
+            for user in blocked_users:
+                username = user.get('username')
+                user_email = user.get('email')
+                blocked_emails.append(username)
+                if user_email and user_email != username:
+                    blocked_emails.append(user_email)
+            ids_to_delete = _get_contact_ids_for_emails(set(blocked_emails))
+            num_deleted = sum([_delete_hubspot_contact(vid) for vid in ids_to_delete])
```

Comment:
I didn't know you could use `sum` on booleans. Cool.

Reply:
nit
```suggestion
            num_deleted = sum(_delete_hubspot_contact(vid) for vid in ids_to_delete)
```

Reply:
done @millerdev